### PR TITLE
Refactors /area & refactors & adds some Ambience... and lighting optimisation...

### PR DESCRIPTION
**Pull request by SamHPurp**
WIP. This isn't done yet. Some more tidying of Area to be done, and there is a hang I need to locate and solve.

:cl: Purpose
fix: Cleans up large parts of area definitions.
add: Adds new ambience sounds (make sure you have Ambience enabled in preferences!)
/:cl:

### CODE REVIEW

File: code/game/area/Dynamic areas.dm [hunk 1]
```diff
@@ -4,7 +4,7 @@
 	var/match_tag = "none"
 	var/match_width = -1
 	var/match_height = -1
-	var/enable_lights = 0
+	var/enable_lights = TRUE
```
Comment (SamHPurp) [inline code/game/area/Dynamic areas.dm]: dumb mistake number 1

**Comment by KasparoVy** [comment]
Yeeees my immersion

**Comment by SamHPurp** [comment]
So in order to do what I want to do... there is something in the lighting code which isn't working right.... so I'm kind of having to improve lighting code in order to get this to work.

this may end up more about lighting optimisation in the end than anything else....

**Comment by SamHPurp** [comment]
@variableundefined test merge please

**Comment by variableundefined** [comment]
Lighting is not my area of expertise, however, there's two very big issue with this PR:
1. Lighting did not initialize at all on a local test server
2. Your optimizations shot CPU use on my local test server sky-high:
Lighting roughly 1 minute in after initialization:
dreamseeker_2018-10-13_10-48-51
CPU profile after initialization, at the same time:
dreamdaemon_2018-10-13_10-48-35
Before optimization, after 2 - 3 minutes:
image

**Comment by variableundefined** [comment]
The relative order should show that the cost of apply_lum and defer were increased massively.

**Comment by SamHPurp** [comment]
Uh. That shouldn't happen, the opposite for me.

Let me check make sure I staged and commited all the right changes.

**Comment by SamHPurp** [comment]
@tigercat2000 Uh. Yes and no. I wasn't *BLINDLY* replacing files. I began to dig deeply into trying to find the problem, and was beginning to *UPDATE* a lot of this code. I then began to roll it back when it realised it was getting really far out of scope.  Clearly I missed some of it.

Chill the fuck out.

**Comment by variableundefined** [comment]
Please reopen the PR when the issue is resolved.

**Pull request closed without merging**